services/ui/.storybook/decorators/ApiConnection.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { action } from '@storybook/addon-actions';
+import { AuthContext } from 'lib/Authenticator';
+import { ApolloClient } from 'apollo-client';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+import { SchemaLink } from 'apollo-link-schema';
+import { ApolloProvider } from 'react-apollo';
+import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools';
+import typeDefs from 'api/src/typeDefs';
```

Comment:
How does this translate to the correct `/services/api` path on disk? In all the other services the way to reference code outside the service folder was to import a module, eg `@lagoon/commons`.

A related question is... since only the `ui` folder is copied into the container, storybook would not be able to run inside docker I presume? Is it only useful to run locally or would we want to consider building storybook inside the containers also?

Reply:
> How does this translate to the correct /services/api path on disk? In all the other services the way to reference code outside the service folder was to import a module, eg @lagoon/commons.

Honestly, I was scratching my head on this one too. PHPStorm rewrote the import path when I was moving files around. And, yet, it still worked fine from my local command line. There's no … config that I saw that referenced the services/api directory, so I have no idea why it actually works.

What's the proper way to import a file from one `/services` directory to another?

> A related question is... since only the ui folder is copied into the container, storybook would not be able to run inside docker I presume?

If we can do a proper import of typedefs and mocks, then we can.

Reply:
What's the _proper_ way? Excellent question. The _current_ way is to add code that needs to be in multiple services to the `@lagoon/commons` package in `node-packages/commons` and that can be imported anywhere it's needed.

Technically the API doesn't use the mocks, so it could go in the ui only. But that seems weird. Open to suggestions. This can also be a follow up issue with the other known issues.